server/routes/app-routes.js

```diff
@@ -0,0 +1,134 @@
+const shopifyFields = require ('../modules/shopify-extras').customerFields;
+
+class AppRoutes {
+
+  constructor(redisClientFactory, dopplerClientFactory, shopifyClientFactory){
+    this.redisClientFactory = redisClientFactory;
+    this.dopplerClientFactory = dopplerClientFactory;
+    this.shopifyClientFactory = shopifyClientFactory;
+  }
+
+  async home(request, response) {
+      const { session: { shop, accessToken } } = request;
+
+      const redis = this.redisClientFactory.createClient();
+      const shopInstance = await redis.getShopAsync(shop, true);
+      
+      // TODO: should we move this to middleware in order to filter all requests?
+      if (shopInstance === null || shopInstance.accessToken !== accessToken) {
+        response.redirect(`/shopify/auth?shop=${shop}`);
+        return;
+      }
+
+      response.render('app', {
+        title: 'Doppler for Shopify',
+        apiKey: process.env.SHOPIFY_APP_KEY,
+        shop: shop,
+        dopplerAccountName: shopInstance.dopplerAccountName
+      });
+  }
+
+  async connectToDoppler(request, response) {
+    const { session: { shop }, body: { dopplerAccountName, dopplerApiKey } } = request;
+    
+    const doppler = this.dopplerClientFactory.createClient(dopplerAccountName, dopplerApiKey);
+    const validCredentials = await doppler.AreCredentialsValidAsync();
+    
+    if (!validCredentials) {
+      response.json({success: false, details: 'Invalid credentials'});
+      return;
+    }
+
+    const redis = this.redisClientFactory.createClient();
+    await redis.storeShopAsync(shop, { dopplerAccountName, dopplerApiKey }, true);
+    response.json({success: true});
```

Comment:
Is it necessary a _success_ field? what do you think about simply relying in status codes?

Reply:
You, sir, are totally right